service/history/timer_queue_active_task_executor.go

```diff
@@ -632,6 +635,28 @@ func (t *timerQueueActiveTaskExecutor) executeWorkflowTimeoutTask(
 	)
 }
 
+func (t *timerQueueActiveTaskExecutor) executeCallbackBackoffTask(ctx context.Context, task *tasks.CallbackBackoffTask) (retErr error) {
+	ctx, cancel := context.WithTimeout(ctx, taskTimeout)
+	defer cancel()
+
+	wfCtx, release, ms, err := t.getValidatedMutableStateForTask(
+		ctx, task, func(ms workflow.MutableState) error {
+			return t.validateCallbackTask(ms, task)
+		},
+	)
+	if err != nil {
+		return nil
+	}
+	defer func() { release(retErr) }()
+
+	// Callback is validated above and must be present in map.
+	callback := ms.GetExecutionInfo().GetCallbacks()[task.CallbackID]
+	if err := callbacks.TransitionRescheduled.Apply(callback, callbacks.EventRescheduled{}, ms); err != nil {
+		return err
+	}
+	return wfCtx.UpdateWorkflowExecutionAsActive(ctx, t.shardContext)
```

Comment:
Can you double check if the underlying implementation will update current record? I remember it uses updateModeUpdateCurrent. If so, then this can't work when workflow is closed and there's a newer run. There will be some runID mismatch error.

Reply:
Oh I didn't know about this detail.
I'll look into it.